ClosureCompilerRecursive.java

```diff
@@ -0,0 +1,123 @@
+   protected boolean writeFiles = true;
+
+   protected String compile(File file) {
+      Compiler compiler = new Compiler();
+
+      List<SourceFile> list = null;
+
+      try {
+          list =
+         CommandLineRunner
+         .getBuiltinExterns(CompilerOptions.Environment.BROWSER);
+      } catch (IOException e) {
+          System.out.println(e.getMessage());
+          System.exit(1);
+      }
+
+      list.add(SourceFile.fromFile(file.getPath()));
+      compiler.compile(new ArrayList<SourceFile>(), list, getOptions());
+      errorCount += compiler.getResult().errors.length;
+      return compiler.toSource();
+   }
+
+   public static void main(String[] args) throws IOException {
+      ClosureCompilerRecursive ccr = new ClosureCompilerRecursive();
+      setOptionsFromArgs(ccr, args);
+      ccr.compileDirectory();
+      System.exit(Math.min(ccr.errorCount, 127));
+   }
+
+   protected static void setOptionsFromArgs(ClosureCompilerRecursive ccr, String[] args) {
+      String dir = null;
+      if (args.length == 1) {
+         dir = args[0];
+      } else if (args.length == 2 && args[0].equals("--validate-syntax")) {
+         ccr.validateSyntaxOnly();
+         dir = args[1];
+      } else {
+         printHelp();
+      }
+
+      ccr.setDirectory(Paths.get(dir));
+   }
+
+   protected static void printHelp() {
+      System.out.println("Usage: closure-compiler-recursive [--validate-syntax] /path/to/directory");
+      System.exit(1);
+   }
+
+   public void compileDirectory() throws IOException {
+      for(File file : getFilesFromDirectory()) {
+         String compressedSource = compile(file);
+         if (writeFiles) {
+            writeFile(file, compressedSource);
+         }
+         System.gc();
+      }
+   }
+
+   public void validateSyntaxOnly() {
+      writeFiles = false;
+   }
+
+   public void setDirectory(Path dir) {
+      directory = dir;
+   }
+
+   protected void writeFile(File file, String contents) throws IOException {
+      PrintWriter writer = new PrintWriter(new BufferedWriter(new FileWriter(file.toString())));
+      writer.println(contents);
+      writer.close();
+   }
+
+   protected File[] getFilesFromDirectory() throws IOException {
+      File[] files = Files.find(
+         directory,
+         Integer.MAX_VALUE,
+           (path, fileAttr) -> fileAttr.isRegularFile()
+                               && path.toString().endsWith(".js"))
+         .map((Path path) -> path.toFile())
+         .toArray(File[]::new);
+      Arrays.sort(files, (File f1, File f2) -> (int) (f2.length() - f1.length()));
+      return files;
+   }
+
+   protected CompilerOptions getOptions() {
+      CompilerOptions options = new CompilerOptions();
+
+      // See :
+      // closure-compiler/src/com/google/javascript/jscomp/CompilerOptions.java
+      // lines 2864-2896
+      options.setLanguageIn(LanguageMode.ECMASCRIPT_2015);
+      options.setLanguageOut(LanguageMode.ECMASCRIPT5_STRICT);
```

Comment:
Is [this what you meant to reference](https://github.com/google/closure-compiler/blob/efe342cc96364f3d6df006b4eda27356a3d9b15a/src/com/google/javascript/jscomp/CompilerOptions.java#L2864-L2896)?

Reply:
Sadly, I didn't write this comment, it was copy-pasta'ed. Will fix!